client/telemetry/src/worker.rs

```diff
@@ -57,13 +60,17 @@ pub enum TelemetryWorkerEvent {
 #[derive(Debug)]
 pub struct TelemetryWorker {
 	/// List of nodes with their maximum verbosity level.
-	nodes: Vec<(node::Node<WsTrans>, u8)>,
+	nodes: Vec<(Arc<Mutex<node::Node<WsTrans>>>, u8)>,
```

Comment:
Why do we store them here in mutexes?

Reply:
The workers live on their own (potentially different thread) and share the nodes with other workers. So that makes sense

Reply:
Why do we need multiple workers?

Reply:
I tried really hard to not have any `Arc` or `Mutex` when writing this code. Are you sure you need to share the nodes between workers?

Reply:
In this new refactoring I removed the `Arc<Mutex<_>>`.

The reason why it was there was because we wanted to re-use the connections to send telemetries but since every substrate node had its own worker, I had to introduce the Arc+Mutex to allow sharing the resources among worker tasks.

In this new implementation there is only one telemetry worker and therefore only one owner of the nodes. The worker receives the payload + the id of the substrate node (span id) to process every message. Every message is then dispatched to the appropriate node sink.